File: android/src/main/java/com/vgretailersdk/getProductDetailsRequest.kt
package com.vgretailersdk

import com.google.gson.annotations.SerializedName

class getProductDetailsRequest{
    @SerializedName("category")
    var category: Array<Int> = emptyArray()
    @SerializedName("subCategory")
    var subCategory: Array<Int> = emptyArray()
    @SerializedName("schemeNumber")
    var schemeNumber: String = ""
    @SerializedName("PartNumber")
    var PartNumber: String = ""

    constructor(category: Array<Int>, subCategory: Array<Int>, schemeNumber: String, PartNumber: String) {
        this.category = category
        this.subCategory = subCategory
        this.schemeNumber = schemeNumber
        this.PartNumber = PartNumber
    }

}

File: ios/GetProductDetailsRequest.swift
import Foundation

class GetProductDetailsRequest: Codable {
    var category: [Int]
    var subCategory: [Int]
    var schemeNumber: String
    var partNumber: String

    // Custom initializer to initialize the object with values from the NSDictionary
    init(details: NSDictionary) {
        category = details["category"] as? [Int] ?? []
        subCategory = details["subCategory"] as? [Int] ?? []
        schemeNumber = details["schemeNumber"] as? String ?? ""
        partNumber = details["PartNumber"] as? String ?? ""
    }
}
